### PR TITLE
More unit-test and API code changes.

### PR DESCRIPTION
**Pull request by cableramki**
1. Added test for usecase where seshat is not available.
2. Changed service discovery API to return URl instead of JSON sub-tree.

### CODE REVIEW

File: libseshat/tests/simple.c [hunk 1]
```diff
@@ -131,10 +131,50 @@ void test_all( void )
...
+    int cnt = 0;
+    char test_buffer[64];
+    char test_url[256];
+    char test_service[32];
+
+    sprintf(test_buffer, "WebPa01");
+    url = seshat_discover(test_buffer);
+    CU_ASSERT(NULL == url);
+    free(url);
+
```
Comment (manderiasian) [inline libseshat/tests/simple.c]: There was a loop here, so now this is only using cnt =0 ?

File: libseshat/tests/simple.c [hunk 2]
```diff
@@ -91,9 +91,9 @@ void test_all( void )
             char test_service[32];
 
             sprintf(test_buffer, "WebPa1%d", cnt);
-            response = seshat_discover(test_buffer);
-            CU_ASSERT(NULL == response);
-            free(response);
+            url = seshat_discover(test_buffer);
+            CU_ASSERT(NULL == url);
```
Comment (cableramki) [inline libseshat/tests/simple.c]: This is a different test, _test_none_. So, this is expected behavior. _test_all_ is still present.

File: libseshat/tests/simple.c (hunk 1 above)
Comment (cableramki) [inline libseshat/tests/simple.c]: This is a different test, _test_none_. So, this is expected behavior. _test_all_ is still present.

File: libseshat/src/libseshat.c [hunk 3]
```diff
@@ -100,16 +103,28 @@ int seshat_register( const char *service, const char *url )
...
     char *response = NULL;
+    char *url = NULL;
     
     if (lib_seshat_is_initialized()) {
         if (NULL != (response = discover_service_data(service))) {
-                errno = 0; 
+            char *substr = strstr(response, URL_SUBSTR_MARKER);
+            substr += sizeof(URL_SUBSTR_MARKER);
+            if (NULL != substr) {
+                char *end = strstr(substr, "\"}");
```
Comment (cableramki) [inline libseshat/src/libseshat.c]: Yes '\' is used as escape here.

File: libseshat/src/libseshat.c [hunk 4]
```diff
@@ -100,16 +103,28 @@ int seshat_register( const char *service, const char *url )
 char *seshat_discover( const char *service )
 {
     char *response = NULL;
+    char *url = NULL;
     
     if (lib_seshat_is_initialized()) {
         if (NULL != (response = discover_service_data(service))) {
-                errno = 0; 
+            char *substr = strstr(response, URL_SUBSTR_MARKER);
+            substr += sizeof(URL_SUBSTR_MARKER);
```
Comment (cableramki) [inline libseshat/src/libseshat.c]: Good catch. Thanks.

**Pull request merged**